src/js/views/mainView.ts

```diff
@@ -0,0 +1,143 @@
+import { $, $$, emit } from '../utils/common';
+import { mainTemplate } from '../templates/main/mainTemplate';
+import { CUSTOM_EVENT, FIRST_INDEX, URL } from '../constants/appContants';
+import { SELECTOR } from '../constants/viewConstants';
+import ManageItemView from './main/mangeItemView';
+import ChargeMoneyView from './main/chargeMoneyView';
+import PurchaseItemView from './main/purchaseItemView';
+import VendingMachine from '../vendingMachine/vendingMachine';
+import UserMenuView from './main/userMenuView';
+import Storage from '../api/storage';
+
+export default class MainView {
+  private $app: HTMLElement;
+  private vendingMachine: VendingMachine;
+  private manageItemView: ManageItemView;
+  private chargeMoneyView: ChargeMoneyView;
+  private purchaseItemView: PurchaseItemView;
+  private userMenuView: UserMenuView;
+
+  constructor() {
+    this.$app = $(SELECTOR.ID.APP);
+    this.vendingMachine = new VendingMachine();
+    this.manageItemView = new ManageItemView(this.vendingMachine);
+    this.chargeMoneyView = new ChargeMoneyView(this.vendingMachine);
+    this.purchaseItemView = new PurchaseItemView(this.vendingMachine);
+    this.userMenuView = new UserMenuView();
+  }
+
+  async render() {
+    this.$app.replaceChildren();
+    this.$app.insertAdjacentHTML('beforeend', mainTemplate);
+    this.checkSignIn();
+
+    $(SELECTOR.CLASS.NAV_CONTAINER).addEventListener('click', this.handleClickNavButton.bind(this));
+    $(SELECTOR.ID.SIGN_BUTTON).addEventListener('click', this.handleSignButtonClick.bind(this));
+
+    await this.vendingMachine.updateResponseItems();
+    await this.vendingMachine.updateResponsCoins();
+  }
+
+  renderPageSection(url: string) {
+    switch (url) {
+      case URL.MANAGE_ITEM:
+        this.manageItemView.render();
+        this.changeButtonColor(SELECTOR.ID_STRING.ITEM_MANGE_TAB);
+        break;
+      case URL.CHARGE_MONEY:
+        this.chargeMoneyView.render();
+        this.changeButtonColor(SELECTOR.ID_STRING.MONEY_CHARGE_TAB);
+        break;
+      case URL.PURCHASE_ITEM:
+        this.purchaseItemView.render();
+        this.changeButtonColor(SELECTOR.ID_STRING.ITEM_PURCHASE_TAB);
+        break;
+      default:
+        this.manageItemView.render();
+        this.changeButtonColor(SELECTOR.ID_STRING.ITEM_MANGE_TAB);
+    }
+  }
```

Comment:
```suggestion
  renderPageSection(url: string) {
    const initialSection = {
      currentView: this.manageItemView,
      buttonId: SELECTOR.ID_STRING.ITEM_MANGE_TAB,
    };

    const section = {
      [URL.MANAGE_ITEM]: initialSection,
      [URL.CHARGE_MONEY]: {
        currentView: this.chargeMoneyView,
        buttonId: SELECTOR.ID_STRING.MONEY_CHARGE_TAB,
      },
      [URL.PURCHASE_ITEM]: {
        currentView: this.purchaseItemView,
        buttonId: SELECTOR.ID_STRING.ITEM_PURCHASE_TAB,
      },
    };

    const { currentView, buttonId } = section[url] ?? initialSection;

    currentView.render();
    this.changeButtonColor(buttonId);
  }
```

당장은 조금 더 길어보일 수 있는데 이렇게 관리하면 view가 추가될수록 더 유지보수하기 쉬워질거같아요 🙂

Reply:
switch문에서 반복되는 구문이 고민이었는데 감사합니다🙂